This patch has been developed inhouse. This has been submitted to the 
community but hasn't been accepted yet. Details at
https://sourceforge.net/p/net-snmp/bugs/2802/
--- net-snmp.old/agent/mibgroup/mibII/at.c	Fri Apr 25 13:46:45 2008
+++ net-snmp/agent/mibgroup/mibII/at.c	Thu Jul 20 14:34:30 2017
@@ -440,9 +440,9 @@
                 break;          /* no need to search further */
             }
         } else {
-            if (snmp_oid_compare(current, olength, name, *length) > 0
-                && snmp_oid_compare(current, olength, lowest,
-                                    *length) < 0) {
+            if ((snmp_oid_compare(current, olength, name, *length) > 0)
+                && ((Found == 0) || (snmp_oid_compare(current, olength, lowest,
+                                    olength) < 0))) {
                 /*
                  * if new one is greater than input and closer to input than
                  * previous lowest, and is not equal to it, save this one as the "next" one.
